openedx/core/djangoapps/content_libraries/models.py

```diff
@@ -0,0 +1,82 @@
+"""
+Models for new Content Libraries
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+import six
+
+User = get_user_model()
+
+
+@six.python_2_unicode_compatible  # pylint: disable=model-missing-unicode
+class ContentLibrary(models.Model):
+    """
+    A Content Library is a collection of content (XBlocks and/or static assets)
+
+    All actual content is stored in Blockstore, and any data that we'd want to
+    transfer to another instance if this library were exported and then
+    re-imported on another Open edX instance should be kept in Blockstore. This
+    model in the LMS should only be used to track settings specific to this Open
+    edX instance, like who has permission to edit this content library.
+    """
+    # Every Library is uniquely and permanently identified by a slug that is set
+    # during creation/import. The slug will appear in the library's opaque key:
+    # e.g. "lib:slug" is the opaque key for a library wtih slug "slug"
+    slug = models.SlugField(primary_key=True)
+    bundle_uuid = models.UUIDField(unique=True, null=False)
+
+    # TODO: Who is authorized to work on this library (in addition to permission inherited from its collection)
+    # -> Define a CollectionPermissions model with m2m field to AuthorizedUser(user_id, role: Admin/Author/Collaborator)
+
+    # How is this library going to be used?
+    allow_public_learning = models.BooleanField(
+        default=False,
+        help_text=("""
+            Allow any user (even unregistered users) to view and interact with
+            content in this library (in the LMS; not in Studio). If this is not
+            enabled, then the content in this library is not directly accessible
+            in the LMS, and learners will only ever see this content if it is
+            explicitly added to a course. If in doubt, leave this unchecked.
+        """),
+    )
+    allow_public_read = models.BooleanField(
+        default=False,
+        help_text=("""
+            Allow any user with Studio access to view this library's content in
+            Studio, use it in their courses, and copy content out of this
+            library. If in doubt, leave this unchecked.
+        """),
+    )
+
+    authorized_users = models.ManyToManyField(User, through='ContentLibraryPermission')
+
+    def __str__(self):
+        return "ContentLibrary ({})".format(self.slug)
+
+    class Meta:
+        verbose_name_plural = "Content Libraries"
+
+
+@six.python_2_unicode_compatible  # pylint: disable=model-missing-unicode
+class ContentLibraryPermission(models.Model):
```

Comment:
Have you seen the [bridgekeeper ADR](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/docs/decisions/0003-permissions-via-bridgekeeper.rst) in edx-platform?

Reply:
Yes, and we have been using bridgekeeper to implement permissions in the LabXchange backend, and I really like it.

I haven't implemented permission checks yet in this PR, and was definitely planning to use Bridgekeeper when I did. But my assumption is that regardless of whether or not one uses bridgekeeper to define the permissions, a data model somewhere is still needed to hold the list of assigned roles etc. right?

Reply:
Yes, definitely. I just wanted to make sure you knew about it, since ADRs are significantly less well advertised than OEPs.